utils.go

```diff
@@ -87,12 +88,15 @@ func getRootlessRuntimeDir(rootlessUID int) (string, error) {
 	if err == nil {
 		return runtimeDir, nil
 	}
-	tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
-	st, err := system.Stat(tmpDir)
-	if err == nil && int(st.UID()) == os.Getuid() && st.Mode()&0700 == 0700 && st.Mode()&0066 == 0000 {
-		return tmpDir, nil
+	initCommand, err := ioutil.ReadFile("/proc/1/comm")
+	if err == nil && string(initCommand) == "systemd" {
```

Comment:
this must be: `err != nil || string(initCommand) == "systemd"`

Reply:
If there is no error and systemd than using `/run/user/X`. I can't understand why `err != nil || _ == "systemd"`. `err != nil` looks like system is unknown, I don't know why we have to use `/run/user/X` in unknown system.

Reply:
`err != nil` means there was an error while trying to detect it, we still want to default to systemd in such case

Reply:
Yes, that's the question. Why we should try to use systemd path if we can't detect system?

Reply:
I agree with @giuseppe, we believe /run is more secure then /tmp. So verifying that the init system is NOT systemd is the only way we should use /tmp.  If we can not verify it is not systemd we assume it is, which is probably a safe assumption.

Reply:
ok, updated